server/src/main/java/com/objectcomputing/checkins/services/skills/SkillController.java

```diff
@@ -1,135 +1,137 @@
+    private SkillServicesImpl skillServices;
 
-    public void setSkillServices(SkillServices skillServices) {
+    public void setSkillServices(SkillServicesImpl skillServices) {
         this.skillServices = skillServices;
     }
 
+    @Error(exception = SkillBadArgException.class)
+    public HttpResponse<?> handleBadArgs(HttpRequest<?> request, SkillBadArgException e) {
+        JsonError error = new JsonError(e.getMessage())
+                .link(Link.SELF, Link.of(request.getUri()));
+
+        return HttpResponse.<JsonError>badRequest()
+                .body(error);
+    }
+
+    @Error(exception = SkillAlreadyExistsException.class)
+    public HttpResponse<?> handleAlreadyExists(HttpRequest<?> request, SkillAlreadyExistsException e) {
+        JsonError error = new JsonError(e.getMessage())
+                .link(Link.SELF, Link.of(request.getUri()));
+
+        return HttpResponse.<JsonError>status(HttpStatus.CONFLICT).body(error);
+    }
+
     /**
      * Create and save a new skill.
      *
-     * @param skill
-     * @return
+     * @param skill, {@link SkillCreateDTO}
+     * @return {@link HttpResponse< Skill >}
      */
 
     @Post(value = "/")
-    public HttpResponse<Skill> createASkill(@Body @Valid Skill skill) {
-        Skill newSkill = skillServices.saveSkill(skill);
+    public HttpResponse<Skill> createASkill(@Body @Valid SkillCreateDTO skill, HttpRequest<SkillCreateDTO> request) {
+        Skill newSkill = skillServices.save(new Skill(skill.getName(), skill.isPending()));
 
         if (newSkill == null) {
-            return HttpResponse.status(HttpStatus.valueOf(409), "already exists");
+            return HttpResponse.status(HttpStatus.valueOf(String.valueOf(HttpStatus.CONFLICT)), "already exists");
         } else {
             return HttpResponse
                     .created(newSkill)
-                    .headers(headers -> headers.location(location(newSkill.getSkillid())));
+                    .headers(headers -> headers.location(
+                            URI.create(String.format("%s/%s", request.getPath(), newSkill.getSkillid()))));
         }
     }
 
-    /**
-     * Load the current skills into checkinsdb.
-     *
-     * @param skillslist
-     * @return
-     */
-
-    @Post("/loadskills")
-    @Consumes(MediaType.APPLICATION_JSON)
-    public void loadSkills(@Body Skill[] skillslist) {
-
-        skillServices.loadSkills(skillslist);
-
-    }
-
-    /**
-     * Get all Skills
-     *
-     * @return {@link Set < Skill >}
-     */
-    @Get("/all")
-    public Set<Skill> readAll() {
-        return skillServices.readAll();
-    }
-
     /**
      * Find and read a skill given its id.
      *
-     * @param skillid
+     * @param skillid {@link UUID} of the skill entry
      * @return
      */
 
     @Get("/{skillid}")
-    public Skill getById(UUID skillid) {
+    public Skill getById(@NotNull UUID skillid) {
         Skill found = skillServices.readSkill(skillid);
         return found;
```

Comment:
can be written as `return skillServices.readSkill(skillid);`

Reply:
done